### PR TITLE
[SPARK-10973][ML][PYTHON] __gettitem__ method throws IndexError exception when we…

### PR DESCRIPTION
**Pull request by zero323**
**gettitem** method throws IndexError exception when we try to access index after the last non-zero entry

```
from pyspark.mllib.linalg import Vectors
sv = Vectors.sparse(5, {1: 3})
sv[0]
## 0.0
sv[1]
## 3.0
sv[2]
## Traceback (most recent call last):
##   File "<stdin>", line 1, in <module>
##   File "/python/pyspark/mllib/linalg/__init__.py", line 734, in __getitem__
##     row_ind = inds[insert_index]
## IndexError: index out of bounds
```

### CODE REVIEW

**Comment by jkbradley** [comment]
ok to test

**Comment by jkbradley** [comment]
test this please

File: python/pyspark/mllib/linalg/__init__.py [hunk 1]
```diff
@@ -770,6 +770,9 @@ def __getitem__(self, index):
             raise ValueError("Index %d out of bounds." % index)
 
         insert_index = np.searchsorted(inds, index)
+        if insert_index >= self.indices.size:
```
Comment (jkbradley) [inline python/pyspark/mllib/linalg/__init__.py]: May as well use ```inds``` here for clarity since that's what is used elsewhere

**Comment by jkbradley** [comment]
LGTM

Ping @mengxr   FYI.  Also, which Spark versions are we patching?

**Comment by jkbradley** [comment]
@zero323 Can you please add tags "[ML] [PYTHON]" to the title of this PR?

**Comment by zero323** [comment]
@jkbradley Done.

**Comment by jkbradley** [comment]
Thanks!  I'll merge this with master once tests pass.

Would you be able to send PRs against branch-1.3, branch-1.4, branch-1.5 in order to backport this to previous Spark versions?  They can use the same JIRA number.

**Comment by jkbradley** [comment]
merging with master

**Comment by zero323** [comment]
@jkbradley https://github.com/apache/spark/pull/9062, https://github.com/apache/spark/pull/9063, https://github.com/apache/spark/pull/9064

**Comment by zero323** [comment]
It should be possible to push this check before binary search: 8a695fe2c3344acd19279fcd539177426d436a02

**Comment by jkbradley** [comment]
@zero323 Good point, that's better for sure.  Do you mind preparing a patch for 1.6 for that?  (I don't think it's necessary to backport it everywhere.)

**Comment by zero323** [comment]
@jkbradley Sure, I can do it later this week. Should I open a new JIRA for that?

**Comment by jkbradley** [comment]
Yes, please.  Thanks!

**Comment by zero323** [comment]
@jkbradley It's my pleasure. I've created a [JIRA](https://issues.apache.org/jira/browse/SPARK-11084) and opened a [PR](https://github.com/apache/spark/pull/9098).

**Pull request closed without merging**